### PR TITLE
Recipe class ks

### LINKED ISSUE #4: Recipes
Users should be able to view a list of recipes. Specifically:

- [ ] As a user, I should be able to filter recipes by type / tag
- [ ] As a user, I should be able to search recipes by ingredients

A recipe should hold on to all its information (provided in the data file). It should be able to:

- [ ] Get the cost of its ingredients
- [ ] Get its directions / instructions

### PR DESCRIPTION
**Pull request by krogowsk531**
# [name/names of contributors]

- [ ] checked functionality of the webpage
- [ ] received help from a mentor/collaborator?
- [ ] added teammates as reviewers
- [x] worked with teammates to resolve Calculate Ingredients Cost Method

## Notes
* Restructured the method in order to return the accurate calculating of ingredients
* All Recipe methods now working
*All tests passing

## Issues Addressed
* Closes #4
* Addresses #<ticket number>

## Questions
*

## Problems or Challenges
* Deciding on correct structure to test ingredients cost

## Screenshots (if applicable)
*

### CODE REVIEW

File: test/recipe-test.js [hunk 1]
```diff
@@ -2,50 +2,259 @@ import { expect } from 'chai';
...
-  let recipe;
-  let recipeInfo;
+  let recipe1;
 
   beforeEach(function() {
-
-    recipeInfo = data[0];
-    recipe = new Recipe(recipeInfo);
-  })
+    recipe1 = new Recipe({
```
Comment (GreyMatteOr) [inline test/recipe-test.js:12]: In the future, can we make up our own data? This is really long!

File: src/recipe.js [hunk 2]
```diff
@@ -7,12 +7,22 @@ class Recipe {
...
     this.tags = recipe.tags;
     this.ingredients = recipe.ingredients;
+    this.instructions = recipe.instructions
   }
-  calculateIngredientsCost() {
-    return this.ingredients.map(i => {
-      ingredientData.find(ingredient => ingredient === i);
-    });
+
+  calculateIngredientsCost(ingredientData) {
```
Comment (GreyMatteOr) [inline src/recipe.js:13]: If `this.ingredients` is an empty array does this return 0? Probably something that we can hardcode to handle as an exception pretty easily. Maybe we can add a sad-path test in the future, also.

**Review by GreyMatteOr — APPROVED** [review]
Nice work!

File: test/recipe-test.js (hunk 1 above)
Comment (kathrynljackson) [reply test/recipe-test.js:12]: for testing, we may be able to create a shorter version that contains the same properties, just less quantity.

**Review by kathrynljackson — APPROVED** [review]

**Pull request merged**